### PR TITLE
Shell for narrative and charts

### PR DESCRIPTION
**Pull request by elayabharath**
Screenshot 2019-12-11 at 12 30 28 PM

Screenshot 2019-12-11 at 12 30 35 PM

### CODE REVIEW

**Comment by lukechurch** [comment]
@marianamarasoiu - could you take a look at this, specifically the use of the Dart code for managing the HTML etc?

**Comment by lukechurch** [comment]
@lukechurch Will update the colours once I finish up the chart visualisation. I'm not too sure how many colours would go into the chart itself, so will update the shell accordingly if that is okay!

**Review by lukechurch — APPROVED** [review]
Thanks @elayabharath - it's an interesting idea to have a narrative shell like this, I'd not thought we'd do that as opposed to having a single page that contains a single visualisation

File: lib/app.dart [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        case html.KeyCode.LEFT:
+          _goToSlide(--currentPage);
+          break;
+        default:
+      }
+    });
+  }
+
+  void _goToSlide(slideNum) {
+    if (slideNum < 0 || slideNum > maxPageCount - 1) {
```
Comment (marianamarasoiu) [inline lib/app.dart]: `slideNum >= maxPageCount`?

File: lib/app.dart [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  void _updateButtonsState() {
+    if (currentPage <= 0) {
+      prevButton.disabled = true;
+    } else {
+      prevButton.disabled = false;
+    }
+
+    if (currentPage >= maxPageCount - 1) {
```
Comment (marianamarasoiu) [inline lib/app.dart]: Doing something a comparison with a -1 is a bit confusing to read and reminds me of the nastiness of off-by-one errors. May I suggest creating a getter for a `firstPageIndex` that just returns 0 and a `lastPageIndex` that does the `maxPageCount - 1` for you? If you do that, feel free to ignore my previous comment in order to keep the -1.

File: lib/app.dart [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+html.DivElement get content => html.querySelector("#content");
+html.DivElement get container => html.querySelector("#container");
+html.DivElement get interactive => html.querySelector("#interactive");
+html.DivElement get progress => html.querySelector("#progress");
+html.ButtonElement get prevButton => html.querySelector("#prev-button");
+html.ButtonElement get nextButton => html.querySelector("#next-button");
+
+class App {
+  int currentPage = 0;
+  int maxPageCount = 10;
```
Comment (marianamarasoiu) [inline lib/app.dart]: If you're making all of the other methods private, you may want to make these instance variables private too.

**Review by marianamarasoiu — APPROVED** [review]
Dart code for HTML looks great, just a couple of tiny nits. Thanks @elayabharath !

File: lib/app.dart (hunk 2 above)
Comment (elayabharath) [reply lib/app.dart]: Ah first and last pageIndex is much cleaner. Have changed it, will make a note for the future as well!

File: lib/app.dart (hunk 3 above)
Comment (elayabharath) [reply lib/app.dart]: Ah fixed

File: lib/app.dart (hunk 1 above)
Comment (elayabharath) [reply lib/app.dart]: Fixed with the pageIndex

**Comment by elayabharath** [comment]
Thanks @lukechurch and @marianamarasoiu

**Pull request merged**